src/coordinatesystems.jl

```diff
@@ -17,44 +19,97 @@ function Polar(r, θ)
+
+@inline Base.convert(::Type{Polar}, p::PolarType) = Polar(p.r, angle(p))
+@inline Base.convert(::Type{Polard}, p::Polar) = Polard(p.r, rad2deg(angle(p)))
+
 
 "`PolarFromCartesian()` - transformation from `AbstractVector` of length 2 to `Polar` type"
-struct PolarFromCartesian <: Transformation; end
-"`CartesianFromPolar()` - transformation from `Polar` type to `SVector{2}` type"
+struct PolarFromCartesian{PT<:PolarType} <: Transformation; end
+PolarFromCartesian() = PolarFromCartesian{Polar}() # default is Polar
+"`PolardFromCartesian()` - transformation from `AbstractVector` of length 2 to `Polar` type"
+const PolardFromCartesian = PolarFromCartesian{Polard}
+
+"`CartesianFromPolar()` - transformation from `Polar` or `Polard` type to `SVector{2}` type"
 struct CartesianFromPolar <: Transformation; end
 
-Base.show(io::IO, trans::PolarFromCartesian) = print(io, "PolarFromCartesian()")
+Base.show(io::IO, trans::PolarFromCartesian{P}) where {P} = print(io, "PolarFromCartesian{$P}()")
 Base.show(io::IO, trans::CartesianFromPolar) = print(io, "CartesianFromPolar()")
 
-function (::PolarFromCartesian)(x::AbstractVector)
+function (::PolarFromCartesian{Polar})(x::AbstractVector)
     length(x) == 2 || error("Polar transform takes a 2D coordinate")
 
     Polar(hypot(x[1], x[2]), atan(x[2], x[1]))
 end
 
-function transform_deriv(::PolarFromCartesian, x::AbstractVector)
+function (::PolarFromCartesian{Polard})(x::AbstractVector)
+    length(x) == 2 || error("Polard transform takes a 2D coordinate")
+
+    Polard(hypot(x[1], x[2]), rad2deg(atan(x[2], x[1])))
+end
+
+function transform_deriv(::PolarFromCartesian{Polar}, x::AbstractVector{T}) where {T}
     length(x) == 2 || error("Polar transform takes a 2D coordinate")
 
     r = hypot(x[1], x[2])
     f = x[2] / x[1]
-    c = one(eltype(x))/(x[1]*(one(eltype(x)) + f*f))
+    c = one(T) / (x[1] * (one(T) + f*f))
+    @SMatrix [ x[1]/r    x[2]/r ;
+              -f*c       c      ]
+end
+function transform_deriv(::PolarFromCartesian{Polard}, x::AbstractVector{T}) where {T}
+    length(x) == 2 || error("Polard transform takes a 2D coordinate")
+
+    r = hypot(x[1], x[2])
+    f = x[2] / x[1]
+    c = rad2deg(one(T)) / (x[1] * (one(T) + f*f))
     @SMatrix [ x[1]/r    x[2]/r ;
               -f*c       c      ]
 end
 transform_deriv_params(::PolarFromCartesian, x::AbstractVector) = error("PolarFromCartesian has no parameters")
 
-function (::CartesianFromPolar)(x::Polar)
-    s,c = sincos(x.θ)
+function (::CartesianFromPolar)(x::PolarType)
+    s,c = sincos(angle(x))
     SVector(x.r * c, x.r * s)
 end
 function transform_deriv(::CartesianFromPolar, x::Polar)
-    sθ, cθ = sincos(x.θ)
+    sθ, cθ = sincos(angle(x))
     @SMatrix [cθ  -x.r*sθ ;
               sθ   x.r*cθ ]
 end
-transform_deriv_params(::CartesianFromPolar, x::Polar) = error("CartesianFromPolar has no parameters")
+function transform_deriv(::CartesianFromPolar, x::Polard)
+    sθ, cθ = sincos(angle(x))
+    a = deg2rad(x.r)
```

Comment:
`deg2rad(x.r)` looks fishy

Reply:
haha yes it does, it's equivalent to

```
deg2rad(1) * x.r
```
but I think if I factor out a little [like you mention](https://github.com/JuliaGeometry/CoordinateTransformations.jl/pull/72#discussion_r545100615) I can clean it up and make it much more clear.